imports/ui/QuestionEditItem.jsx

```diff
@@ -378,31 +378,37 @@ export class QuestionEditItem extends Component {
    * Calls {@link module:questions~"questions.insert" questions.insert} to save question to db
    */
   saveQuestion () {
-    const user = Meteor.user()
-    let question = _.extend({
-      createdAt: new Date(),
-      approved: user.hasGreaterRole('professor') || user.isInstructor(this.props.courseId),
-    }, _.omit(this.state, 'courses'))
-
-    if (question.options.length === 0 && question.type !== QUESTION_TYPE.SA) return
-
-    if (this.props.sessionId) question.sessionId = this.props.sessionId
-    if (this.props.courseId) question.courseId = this.props.courseId
-
-    // insert (or edit)
-    Meteor.call('questions.insert', question, (error, newQuestion) => {
-      if (error) {
-        alertify.error('Error: ' + error.error)
-      } else {
-        if (!this.state._id) {
-          alertify.success('Question Saved')
-          if (this.props.onNewQuestion) this.props.onNewQuestion(newQuestion._id)
-        } else {
-          alertify.success('Edits Saved')
-        }
-        this.setState(newQuestion)
+
+    Meteor.call('courses.getCourseApproved', this.props.courseId, (err, approved) => {
+      if (err) alertify.error('Cannot get question permissions')
+      else {
+        const user = Meteor.user()
+        let question = _.extend({
+          createdAt: new Date(),
+          approved: user.hasGreaterRole('professor') || user.isInstructor(this.props.courseId) || approved,
+        }, _.omit(this.state, 'courses'))
+
```

Comment:
In general, I think we should move the question from this.state to this.state.question, as it's a bit confusing for the state of QuestionEditItem to have properties that belong to the question being edited as well as properties about the component itself... Make this an issue for a small refactor...